entrypoint/entrypoint/orchestrator.py

```diff
@@ -441,3 +451,9 @@ def require_true(expr: bool, msg: str):
     if not expr:
         logging.error(msg)
         exit(1)
+
+def is_valid_container_platform(img_platform):
+    # regex for detecting 'os/cpu/variant'
+    # os/cpu are required whereas variant is optional
+    pattern = r'^[^/]+/[^/]+(?:/[^/]+)?$'
+    return bool(re.match(pattern, img_platform))
```

Comment:
Since this regex is used to check if platform string satisfies regex pattern or not, non-capturing group(`?:`) is not needed here. 
```suggestion
    pattern = r'^[^/]+/[^/]+(/[^/]+)?$'
    return bool(re.match(pattern, img_platform))
```


Although capturing group is not needed here, other technique I'd like to share is to use named capturing group just for readability.

```
pattern = r'^(?P<os>[^/]+)/(?P<cpu>[^/]+)(?P<variant>/[^/]+)?$'
```

By using named capturing group, you can name group of regex like this example.

Reply:
This is great, I didn't know you could name capture groups in REGEX.
I'll use that going forward.